src/kaleidoscope/device/dygma/Raise.cpp

```diff
@@ -0,0 +1,599 @@
+  {19, 20, 21, 22, 23, 24, 25, XX,      XX, XX,   27 + LKEY, 26 + LKEY, 25 + LKEY, 24 + LKEY, 23 + LKEY, 22 + LKEY},
+  {26, 27, 28, 29, 30, XX, 31, 32,      35 + LKEY, 34 + LKEY, 33 + LKEY, 32 + LKEY, 31 + LKEY, 30 + LKEY, 29 + LKEY, 28 + LKEY},
+};
+
+// maps keyboard led number (0->PROPS_::LED_COUNT-1) to the SLED led number (0->LPH-1 on left side  and LPH to LPH*2-1 on the right side)
+// LPH comes from keyboardioscanner.h - leds per hand = 72 defined by the size of the buffer used to transfer data to sides
+static constexpr uint8_t led_map[2][133] = {
+  // ISO
+  {
+    // left side - 33 keys includes LP
+    0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17, 18, 19, 20, 21, 22, 23, 24, 25, 26, 27, 28, 29, 30, 68, 69,
+
+    // right side - 36 keys includes LP
+    0 + LPH, 1 + LPH, 2 + LPH, 3 + LPH, 4 + LPH, 5 + LPH, 6 + LPH, 7 + LPH, 8 + LPH, 9 + LPH, 10 + LPH, 11 + LPH, 12 + LPH, 13 + LPH, 14 + LPH, 15 + LPH, 16 + LPH, 17 + LPH, 18 + LPH, 19 + LPH,
+    20 + LPH, 21 + LPH, 22 + LPH, 23 + LPH, 24 + LPH, 25 + LPH, 26 + LPH, 27 + LPH, 28 + LPH, 29 + LPH, 30 + LPH, 31 + LPH, 32 + LPH, 33 + LPH, 68 + LPH, 69 + LPH,
+
+    // left under glow - 30
+    34, 35, 36, 37, 38, 39, 40, 41, 42, 43, 44, 45, 46, 47, 48, 49, 50, 51, 52, 53, 54, 55, 56, 57, 58, 59, 60, 61, 62, 63,
+
+    // right underglow - 32
+    34 + LPH, 35 + LPH, 36 + LPH, 37 + LPH, 38 + LPH, 39 + LPH, 40 + LPH, 41 + LPH, 42 + LPH, 43 + LPH, 44 + LPH, 45 + LPH, 46 + LPH, 47 + LPH, 48 + LPH, 49 + LPH, 50 + LPH, 51 + LPH,
+    52 + LPH, 53 + LPH, 54 + LPH, 55 + LPH, 56 + LPH, 57 + LPH, 58 + LPH, 59 + LPH, 60 + LPH, 61 + LPH, 62 + LPH, 63 + LPH, 64 + LPH, 65 + LPH, XX, // last one is the Huble, never send to SLED
+  },
+  // ANSI
+  {
+    // left side - 32 keys includes LP: key 19 is missing for ANSI layout
+    0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17, 18, 20, XX, 21, 22, 23, 24, 25, 26, 27, 28, 29, 30, 68, 69,
+
+    // right side - 36 keys includes LP
+    0 + LPH, 1 + LPH, 2 + LPH, 3 + LPH, 4 + LPH, 5 + LPH, 6 + LPH, 15 + LPH, 8 + LPH, 9 + LPH, 10 + LPH, 11 + LPH, 12 + LPH, 13 + LPH, 14 + LPH, 7 + LPH, 16 + LPH, 17 + LPH, 18 + LPH, 19 + LPH,
+    20 + LPH, 21 + LPH, 22 + LPH, 23 + LPH, 24 + LPH, 25 + LPH, 26 + LPH, 27 + LPH, 28 + LPH, 29 + LPH, 30 + LPH, 31 + LPH, 32 + LPH, 33 + LPH, 68 + LPH, 69 + LPH,
+
+    // left under glow - 30
+    34, 35, 36, 37, 38, 39, 40, 41, 42, 43, 44, 45, 46, 47, 48, 49, 50, 51, 52, 53, 54, 55, 56, 57, 58, 59, 60, 61, 62, 63,
+
+    // right underglow - 32
+    34 + LPH, 35 + LPH, 36 + LPH, 37 + LPH, 38 + LPH, 39 + LPH, 40 + LPH, 41 + LPH, 42 + LPH, 43 + LPH, 44 + LPH, 45 + LPH, 46 + LPH, 47 + LPH, 48 + LPH, 49 + LPH, 50 + LPH, 51 + LPH,
+    52 + LPH, 53 + LPH, 54 + LPH, 55 + LPH, 56 + LPH, 57 + LPH, 58 + LPH, 59 + LPH, 60 + LPH, 61 + LPH, 62 + LPH, 63 + LPH, 64 + LPH, 65 + LPH, XX, // last one is the Huble, never send to SLED
+  }
+};
+
+/********* TWI *********/
+void twi_disable(void) {
```

Comment:
Huh. for the keyboardio library, we don't, iirc, mix the Wire implementation and our low level C library. This feels a little funny.

Reply:
We don't mix them here, either. Raise uses Wire, just wrapped in a TWI dressing. The`twi_disable` and `twi_init` methods are here in Raise.cpp to make it easy to call them, without having to reach for the `TWI` class, or pull in the whole wrapping stuff.

I should probably clean this up, perhaps even transition the Raise to the same TWI implementation the Model01 uses - or something similar.

Reply:
I've removed the wrappers from here. There's still `TWI.h` and `TWI.cpp` in `device/dygma/raise/`, which wrap `Wire`. I'm looking into unwiring those.